src/task.c

```diff
@@ -765,12 +833,17 @@ task_system_create(platform_heap_id    hid,
          *system = NULL;
          return rc;
       }
+      nbg_threads_created += (use_bg_threads ? num_bg_threads[type] : 0);
    }
 
-   task_run_thread_hooks(ts);
-   const threadid tid      = platform_get_tid();
-   ts->thread_scratch[tid] = ts->init_task_scratch;
-
+   // Wait-for all background threads to start-up and register. This is not
+   // really needed right now, but it allows for reliable unit-testing to verify
+   // state of the task system when bg threads are configured.
+   threadid *max_tid = task_system_get_max_tid(ts);
```

Comment:
Either move this into the test or implement it by having each thread increment a counter and looping here until the counter reaches the expected value.

But please no sleeps.

Reply:
Removed this from here and moved the logic to a helper function, `create_task_system_with_bg_threads()` defined in test code. That fn will loop around a bit sleeping till all configured bg-threads are started up.